src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RijndaelManagedTransform.Browser.cs

```diff
@@ -0,0 +1,1360 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Security.Cryptography
+{
+    internal enum RijndaelManagedTransformMode
+    {
+        Encrypt = 0,
+        Decrypt = 1
+    }
+
+    internal sealed class RijndaelManagedTransform : ICryptoTransform, ILiteSymmetricCipher
+    {
+        private CipherMode m_cipherMode;
+        private PaddingMode m_paddingValue;
+        private RijndaelManagedTransformMode m_transformMode;
+
+        private int m_blockSizeBytes;
+        private int m_inputBlockSize;
+        private int m_outputBlockSize;
+
+        private int[] m_encryptKeyExpansion;
+        private int[] m_decryptKeyExpansion;
+
+        private int m_Nr;
+        private int m_Nb;
+        private int m_Nk;
+
+        private int[] m_encryptindex;
+        private int[] m_decryptindex;
+
+        private int[] m_IV;
+        private int[] m_lastBlockBuffer;
+        private byte[]? m_depadBuffer;
+
+        public RijndaelManagedTransform(byte[] rgbKey,
+                                        CipherMode mode,
+                                        ReadOnlySpan<byte> rgbIV,
+                                        int blockSize,
+                                        int feedbackSize,
+                                        PaddingMode PaddingValue,
+                                        RijndaelManagedTransformMode transformMode)
+        {
+            if (rgbKey == null)
+                throw new ArgumentNullException(nameof(rgbKey));
+
+            Debug.Assert(mode == CipherMode.CBC, "CipherMode is unsupported");
+            Debug.Assert(PaddingValue == PaddingMode.PKCS7, "PaddingMode is unsupported");
```

Comment:
Was this taken from the .NET Framework?

Rijndael in .NET Framework permitted non-standard block sizes. We should `Debug.Assert(blockSize == 16)` (assuming the `blockSize` parameter is passed in bytes. If it's passed in bits, then `Debug.Assert(blockSize == 128)`.

I just want to make sure we don't accidentally start supporting AES with non-AES block sizes.

Reply:
Answering my own question because reading helps...

> This ports the C# implementation of AES from https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/mscorlib/system/security/cryptography/rijndaelmanagedtransform.cs,


So I think asserting the block size would be a good idea (and in any other blockSize inputs).